src/Microsoft.OData.Edm/Csdl/Parsing/AnnotationJsonParser.cs

```diff
@@ -0,0 +1,515 @@
+//---------------------------------------------------------------------
+// <copyright file="AnnotationJsonParser.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+#if NETSTANDARD2_0
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.Text.Json;
+using Microsoft.OData.Edm.Csdl.Parsing.Ast;
+using Microsoft.OData.Edm.Validation;
+using Microsoft.OData.Edm.Vocabularies;
+
+namespace Microsoft.OData.Edm.Csdl.Parsing
+{
+    /// <summary>
+    /// Provides functionalities for parsing CSDL-JSON annotations.
+    /// </summary>
+    internal static class AnnotationJsonParser

```

Comment:
>AnnotationJsonParser [](start = 26, length = 20)

should we create a "Json" folder under "Parsing" for the JSON-specific classes?

Reply:
I did the same thing, But i reverted it because the class name clearly refreshes the usage. 
And the whole number for JSON parsers is 3, "CsdlJsonParser, SchemaJsonParser, AnnotationJsonParser". So, at current stage,  maybe it's fine to keep them in the "Parsing" folder.


---
In reply to: [443009888](https://github.com/OData/odata.net/pull/1775#discussion_r443009888) [](ancestors = 443009888)